src/actix/api/cluster_api.rs

```diff
@@ -79,6 +79,13 @@ fn remove_peer(
             });
         }
 
+        // If we reach here, either there are no shards or force removal is requested
+        if has_shards {
+            log::warn!("Aborting peer transfers for peer {peer_id} before force removal");
+            toc.abort_peer_transfers(dispatcher.clone(), access, peer_id)
+                .await?;
+        }
```

Comment:
IMO, this should be done differently. Instead of explicitly submitting `ShardTransferOperations::Abort` in the API, we should modify the code that is called when `RemovePeer` consensus operation is applied, and make it abort transfers as part of `RemovePeer` operation.